src/main.ts

```diff
@@ -2,8 +2,46 @@
 
 import * as ts from 'typescript';
 
-class Transpiler {
+const DEBUG = true;
+
+export const options: ts.CompilerOptions = {
+  allowNonTsExtensions: true,
+  module: ts.ModuleKind.CommonJS,
+  target: ts.ScriptTarget.ES5,
+};
+
+export class Minifier {
   constructor() {}
 
-  helloWorld() { return 'hello world'; }
-}
\ No newline at end of file
+  visit(node: ts.Node) {
+    switch (node.kind) {
+      case ts.SyntaxKind.PropertyAccessExpression:
+        result = '';
+        var pae = <ts.PropertyAccessExpression>node;
+        result += this.visit(pae.expression);
+        result += '.';
+        // Adds '$mangled' to an identifier to ensure that the identifier is being altered.
+        result += pae.name.text + '$mangled';
+        break;
+
+      default:
+        var text = node.getText();
+        var children = node.getChildren();
+        var result = '';
+        var prevEnd = 0;
+        children.forEach((child) => {
+          var childText = this.visit(child);
```

Comment:
I think this would be easier to read if you'd first append the text between the last end and the start of this child, *then* visit the child and append its text (in a separate statement). That way the flow is more natural.